### ISSUE TITLE
When running python with -OO option, an empty matplotlib docstring causes an exception.

### ISSUE THREAD

**Issue by durian888** [matplotlib/matplotlib#18169]
### Bug report

**Bug summary**

Matplotlib generates an exception in _axes.py when python is started with -OO option because matplotlib code attempts to update an docstring set to None.

**Code for reproduction**
see code snippet below 

issue is in _axes.py 
line 5305
Note I think there may be other locations in the file where format is called on doc string without any checks

```python
#matplotlib_test2.py
#the example 
#https://matplotlib.org/gallery/lines_bars_and_markers/simple_plot.html#sphx-glr-gallery-lines-bars-and-markers-simple-plot-py
#Simple Plot 

#Create a simple plot.

import matplotlib
import matplotlib.pyplot as plt
import numpy as np

# Data for plotting
t = np.arange(0.0, 2.0, 0.01)
s = 1 + np.sin(2 * np.pi * t)

fig, ax = plt.subplots()
ax.plot(t, s)

ax.set(xlabel='time (s)', ylabel='voltage (mV)',
       title='About as simple as it gets, folks')
ax.grid()

fig.savefig("test.png")
plt.show()

```

execute at command line
python matplotlib_test2.py 
gives a nice plot

execute 
python -OO matplotlib_test2.py
Traceback (most recent call last):
  File "matplotlib_test2.py", line 2, in <module>
    import matplotlib.pyplot as plt
  File "C:\Users\djb\work\telestar_py38\.env\lib\site-packages\matplotlib\pyplot.py", line 43, in <module>
    from matplotlib.figure import Figure, figaspect
  File "C:\Users\djb\work\telestar_py38\.env\lib\site-packages\matplotlib\figure.py", line 18, in <module>
    from matplotlib import docstring, projections
  File "C:\Users\djb\work\telestar_py38\.env\lib\site-packages\matplotlib\projections\__init__.py", line 1, in <module>
    from .. import axes, docstring
  File "C:\Users\djb\work\telestar_py38\.env\lib\site-packages\matplotlib\axes\__init__.py", line 1, in <module>
    from ._subplots import *
  File "C:\Users\djb\work\telestar_py38\.env\lib\site-packages\matplotlib\axes\_subplots.py", line 6, in <module>
    from matplotlib.axes._axes import Axes
  File "C:\Users\djb\work\telestar_py38\.env\lib\site-packages\matplotlib\axes\_axes.py", line 71, in <module>
    class Axes(_AxesBase):
  File "C:\Users\djb\work\telestar_py38\.env\lib\site-packages\matplotlib\axes\_axes.py", line 5305, in Axes
    fill_between.__doc__ = _fill_between_x_or_y.__doc__.format(
AttributeError: 'NoneType' object has no attribute 'format'

**Expected outcome**
a nice plot when running python with -OO command line parameter

**Matplotlib version**
windows: 10 64bit
Python 3.8.5 (tags/v3.8.5:580fbb0, Jul 20 2020, 15:57:54) [MSC v.1924 64 bit (AMD64)] on win32
matplotlib version: 3.3.0
matplotlib backend: TkAgg

python from python-3.8.5-amd64.exe from www.python.org
matplot via pip

**Comment by WeatherGod** [comment]
Fairly certain that this got fixed recently in master (but not in the
v3.3.0 release, I don't think).

**Comment by jklymak** [comment]
This is #17970, and will be fixed in 3.3.1
